app/javascript/packages/document-capture/components/acuant-capture-canvas.jsx

```diff
@@ -32,10 +33,16 @@ function AcuantCaptureCanvas() {
   const cameraRef = useRef(/** @type {HTMLDivElement?} */ (null));
 
   useEffect(() => {
+    let originalDescriptor;
+    let canvas;
+
     function onAcuantCameraCreated() {
-      const canvas = document.getElementById('acuant-ui-canvas');
+      canvas = document.getElementById('acuant-ui-canvas');
       // Acuant SDK assigns a callback property to the canvas when it switches to its "Tap to
       // Capture" mode (Acuant SDK v11.4.4, L158). Infer capture type by presence of the property.
+      if (originalDescriptor === undefined) {
+        originalDescriptor = Object.getOwnPropertyDescriptor(canvas, 'callback');
+      }
```

Comment:
We're probably not expecting this to be anything other than `undefined`, though I suppose technically the `acuantcameracreated` callback could be called twice. But if it were, I wouldn't necessarily expect the `defineObservableProperty` on the next line to work. (Maybe this is what's causing the error being logged? 🤔 )

Maybe instead, we should guarantee that the callback is only called once, and that way we can not worry about whether this is `undefined`.

```suggestion
      cameraRef.current?.removeEventListener('acuantcameracreated', onAcuantCameraCreated);
      originalDescriptor = Object.getOwnPropertyDescriptor(canvas, 'callback');
```

Or if we wanted to be able to handle multiple callbacks, we could reuse the clean-up logic you added below?

```ts
function resetObservableProperty() {
  if (originalDescriptor !== undefined) {
    Object.defineProperty(canvas, 'callback', originalDescriptor);
  } else {
    delete canvas?.callback;
  }
}

function onAcuantCameraCreated() {
  resetObservableProperty();
  canvas = document.getElementById('acuant-ui-canvas');
  originalDescriptor = Object.getOwnPropertyDescriptor(canvas, 'callback');
  // ...
}

// ...

return () => {
  cameraRef.current?.removeEventListener('acuantcameracreated', onAcuantCameraCreated);
  resetObservableProperty();
};
```

Reply:
I moved the reset logic to a new method. Adding in debug console.logs (see 0122f20bb), I see an odd pattern of calls:

<img width="385" alt="acuantCaptureCanvas-calls" src="https://github.com/18F/identity-idp/assets/2583060/6670eb03-0bb3-406b-be77-b672f70a6e29">

Everything before the final reset happens when the SDK capture comes up. Upon auto capture, the reset is called and the component is unloaded. The component is created twice, but useEffect is only called once? 

I'm checking the originalDescriptor before I grab it since without it it's getting called twice and thus we're "resetting" it to the descriptor that we defined, not the original.

Reply:
Also, I'm able to call defineProperty again to reset it only because I added [configurable: true](https://github.com/18F/identity-idp/pull/10762/files#diff-4c6533807dd9a52d37bbd4a31cf7d4582e9879f7d160f2e0e4beddb06231b7f6R26) to the original call.

Reply:
I made a branch attempting a cleanup of the observable property here:
https://github.com/18F/identity-idp/compare/margolis-observable-property

(and also put the observable property into its own file)

Reply:
Looks clean to me and tests cleanly (no TypeError raised). I wasn't confident on how to fully reset the property, thus my originalDescriptor jumble. 🙄

My only mental itch would be... should the methods be called `defineObservableProperty` and `removeObservableProperty`? In particular, in your version, we aren't removing the property, just the callback. Seems like they are actually more like `observeProperty` and `stopObservingProperty`.

Do you want to open a PR and have me close this, or shall I pull in your changes?

Reply:
I like the name suggestion! I will make a PR into this branch.

Reply:
Opened https://github.com/18F/identity-idp/pull/10809